tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for the Empirical distribution."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import importlib
+
+# Dependency imports
+import numpy as np
+import tensorflow as tf
+
+from tensorflow_probability.python.distributions import empirical
+from tensorflow.python.framework import test_util
+
+
+def try_import(name):  # pylint: disable=invalid-name
+  module = None
+  try:
+    module = importlib.import_module(name)
+  except ImportError as e:
+    tf.logging.warning("Could not import %s: %s" % (name, str(e)))
+  return module
+
+def entropy(p):
+  q = 1. - p
+  return -q * np.log(q) - p * np.log(p)
+
+def make_empirical(batch_shape, validate_args=False):
+  p = np.random.uniform(size=list(batch_shape))
+  return empirical.Empirical(params=p, validate_args=validate_args)
+
+
+class EmpiricalTest(tf.test.TestCase):
```

Comment:
This looks like a great start on tests! I think we need to expand coverage to test all distribution methods (mean, mode, stddev, entropy, etc), and I'd like to see more testing of multivariate sample spaces (as discussed in a few places above).

We also need to parameterize these tests to cover the cases where parameters have dynamic and static shapes, and different dtypes. See the [unit test](https://github.com/tensorflow/probability/blob/master/UNITTEST.md) guide for an easy formula for doing this.

Reply:
Fixed. I expand test coverage under this framework.